core/src/main/java/io/ultrabrew/metrics/data/DistributionBucket.java

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the Apache License 2.0 license. See LICENSE file in Ultrabrew Metrics
+// for terms.
+
+package io.ultrabrew.metrics.data;
+
+import java.util.function.Predicate;
+
+/**
+ * A distribution bucket specification used for histograms. A distribution bucket is an sorted array
+ * of unique numeric values. Each bucket represents a distribution range of the measurement.
+ *
+ * <P>For a given distribution array: [0, 10, 100, 500, 1000], the buckets would be like:</P>
+ * <ul>
+ * <li>[0,10) for values 0-9</li>
+ * <li>[10,100) for values 10-99</li>
+ * <li>[100,500) for values 100-499</li>
+ * <li>[500,1000) for values 500-999</li>
+ * <li>overflow  for values {@literal >}= 1000</li>
+ * <li>underflow for values {@literal <} 0</li>
+ * </ul>
+ *
+ * @see BasicHistogramAggregator
+ */
+public class DistributionBucket {
+
+  private static final String UNDERFLOW = "underflow";
+  private static final String OVERFLOW = "overflow";
+
+  private final long[] buckets;
+
+  /**
+   * Creates a distribution for given bucket spec.
+   *
+   * @param buckets sorted array of unique values
+   */
+  public DistributionBucket(final long[] buckets) {
+
+    assert buckets.length >= 2 : "Minimum bucket length is 2";
+    assert isSorted(buckets) : "Bucket should be sorted in ascending order";
+    assert !hasDuplicate(buckets) : "Bucket should not have duplicate entries";
```

Comment:
Please don't use `assert` in non-performance-critical code, almost no one ever runs JVM with assertions enabled.